releasenotes/notes/kubernetes-service-names-955b78d3d2b134f6.yaml

```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    Support the Kubernetes service name tags on containers.
+    The tag used is kube_service.
```

Comment:
we should add the known issue that `kube_service` tags don't update with services created after a pod's creation, as it was the case in agent5.
I think it's good enough for 6.0, but we should document it, and create a card for that.


BTW, could you field-test that it does work with pods created after the agent? IIRC pods are added as endpoints only when their liveliness probe is green. If we detect a new container before that, we might never get it's service tag. WDYT?

Reply:
You're right, we need to only detect new container when the Pod resource where it belong is marked as Ready.
The readiness status is reported in the Pod `.status.conditions`.
I made some changes to catch this new constraint.

From my local testing I made the following steps to observe the tagging:
- agent running
- deployment of a new service/deployment (with, without readiness probes)
- tagging is effective on the container after 3-6s without readiness probe, (24-26s with an initialDelaySeconds set to 20).